content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1411 @@
+non-canonical URLs to the canonical URL: _\https://jenkins.acme.example/_
+
+I have installed the IIS (8.5) role using the _Add Roles and Features
+Wizard_ with the all the default and also the following non-default
+features:
+
+* HTTP Redirection (Under _Common HTTP Features_, to
+redirect [.nolink]#http(s)://jenkins/,# etc. to
+https://jenkins.acme.example/)
+* WebSocket Protocol (Under _Application Development_, because I felt
+like it)
+
+Then I installed URL Rewrite and Application Request Routing.
+
+==== Configuration Time
+
+===== Enabling Reverse Proxy functionality
+
+. In the _Internet Information Services (IIS) Manager_ click on the
+VRTJENKINS01 server.
+. Go to _Application Request Routing Cache_.
+. In the _Actions_ panel click on _Server Proxy Settings..._
+. Enable the proxy
+. Disable the _Reverse rewrite host in response header_
+.. Don't worry, it will work, just follow the rest of the instructions
+. Set the _Response buffer threshold (KB)_ to 0.
+.. This helps to prevent HTTP 502 errors on Jenkin's Replay pages.
+. Apply (the _Actions_ panel again)
+
+===== Configuring TLS/SSL
+
+Out of scope, there are enough tutorials on the rest of the interwebs
+for this part.
+The rest of this tutorial will assume it has been configured with a
+certificate trusted by your browser of choice.
+
+===== Configuring rules for response rewriting
+
+. Go to the _Default Web Site_
+. Go to __URL Rewrite__
+. In the _Actions_ panel click _View Server Variables..._
+. Add the following is not already define on the server level:
+.. Name: *HTTP_FORWARDED*
+. Click on _Back to Rules_
+. _Click on Add Rule(s)..._
+. Select _Reverse Proxy_ and click on OK
+. Enter _jenkins.acme.example_ and click on OK
+. Open the rule you just created
+. Under _Conditions_ add:
+.. Condition input: *\{CACHE_URL}*
+.. Pattern: *^(http|ws)s://*
+. Under _Server Variables_ add:
+.. Name: *HTTP_FORWARDED*, Value:
+*for=\{REMOTE_ADDR};by=\{LOCAL_ADDR};host="\{HTTP_HOST}";proto="https"*,
+Replace: yes
+... Jenkins runs under Jetty, Jetty supports
+https://tools.ietf.org/html/rfc7239[RFC7239], so all should be well.
+. Under Action change:
+..  Rewrite URL to
+*\{C:1}[.nolink]#://jenkins.acme.example:8080\{UNENCODED_URL#}*
+... Note that there is no slash between the port number and the opening
+curly bracket
+.. *Remove* the check from the *Append query string* checkbox
+. Apply the changes.
+. Edit _C:\Windows\System32\drivers\etc\hosts_ so that
+*jenkins.acme.example* points to 127.0.0.1
+.. When resolving names Windows will check if the name is its own name
+before consulting the hosts file. Meaning that adding _vrtjenkins01_ or
+_vrtjenkins01.acme.example_ to the hosts file won't have any effect.
+... The hosts file will however be consulted before consulting the DNS
+infrastructure
+
+===== Experiencing the dreaded "It appears that your reverse proxy set up is broken." error for yourself
+
+. [.nolink]#https://jenkins.acme.example/configure#
```

Comment:
The confluence style `nolink` markup should be replaced with simpler asciidoc escape.
```suggestion
. \https://jenkins.acme.example/configure
```